### PR TITLE
Added controllers and jsp pages, Authentication filter. CarDaoImpl an…

### PR DESCRIPTION
**Pull request by oAlExAnDeRo**
…d DriveDaoImpl were changed. Also, web.xml.

### CODE REVIEW

File: src/main/java/taxi/controller/LoginController.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String login = req.getParameter("login");
+        String password = req.getParameter("password");
+        try {
+            Driver driver = authenticationService.login(login, password);
+            HttpSession session = req.getSession();
+            session.setAttribute("driver_id", driver.getId());
+            resp.sendRedirect("/index");
```
Comment (Rommelua) [inline src/main/java/taxi/controller/LoginController.java:37]: context path?

File: src/main/java/taxi/controller/LogoutController.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class LogoutController extends HttpServlet {
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getSession().invalidate();
+        resp.sendRedirect("/index");
```
Comment (Rommelua) [inline src/main/java/taxi/controller/LogoutController.java:14]: context path?

File: src/main/java/taxi/service/AuthenticationServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import taxi.dao.DriverDao;
+import taxi.exception.AuthenticationException;
+import taxi.lib.Inject;
+import taxi.lib.Service;
+import taxi.model.Driver;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private DriverDao driverDao;
```
Comment (Rommelua) [inline src/main/java/taxi/service/AuthenticationServiceImpl.java:13]: image

File: src/main/java/taxi/service/AuthenticationServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+    @Override
+    public Driver login(String login, String password) throws AuthenticationException {
+        Optional<Driver> driver = driverDao.findByLogin(login);
+        if (driver.isEmpty()) {
+            throw new AuthenticationException("Login or password was incorrect");
+        }
+        if (driver.get().getPassword().equals(password)) {
```
Comment (Rommelua) [inline src/main/java/taxi/service/AuthenticationServiceImpl.java:21]: image

**Review by Rommelua — CHANGES_REQUESTED** [review]
Before you create PR with your homework, you need to go through the checklist under the task and correct all the points described there. The mentor will not check the work until the checklist points are corrected

**Review by aaananas — APPROVED** [review]

**Pull request closed without merging**